sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsDiskStore.scala

```diff
@@ -110,25 +122,38 @@ class RapidsDiskStore(diskBlockManager: RapidsDiskBlockManager)
   class RapidsDiskBuffer(
       id: RapidsBufferId,
       fileOffset: Long,
-      size: Long,
+      uncompressedSize: Long,
+      compressedSize: Long,
       meta: TableMeta,
       spillPriority: Long)
       extends RapidsBufferBase(
         id, meta, spillPriority) {
     private[this] var hostBuffer: Option[HostMemoryBuffer] = None
 
-    override val memoryUsedBytes: Long = size
+    override val memoryUsedBytes: Long = uncompressedSize
```

Comment:
It all depends on what it is going to be used for. Technically there is no memory being used, so this API might as well return 0.  The docs for it say.

> The size of this buffer in bytes in its _current_ store. As the buffer goes through
>   contiguous split (either added as a contiguous table already, or spilled to host),
>   its size changes because contiguous_split adds its own alignment padding.
>   
>   NOTE: Do not use this size to allocate a target buffer to copy, always use `getPackedSize.`

The problem is that this API is used in all kinds of places for metrics and logging. The only places I can find where it is used beyond metrics are places where we know exactly what type is being passed in, by convention, so that we end up with a lot of coupling happening.

To me we should file a follow on issue and make it new APIs around what we actually want to work with. I don't know what those APIs are exactly, but a few that come to mind are.

1. how much (disk, host memory, device memory) would freeing/spilling this make available. I can see two options for this.
  a. one API that says how much and another API that says where it is stored.
  b. one API that returns a case class with all three values in it. 
2. how much memory do I need to hold the the uncompressed, packed version of this
3. what codec is this compressed with.
4. is this currently packed

For now I really don't care what we return so long as we have a follow on issue to clean this up.

Reply:
I think it would make sense to expose & cleanup our metadata (we have it as `TableMeta`) to include more specifics. When we start looking at UCX compression (nvcomp or CPU) we could revisit this, but I would like to keep it as a different issue/PR. 

We have I think most of what Bobby wants in the `TableMeta`, that said we need to expose it better and carry it through the whole stack, not just in the shuffle cases. Right now we only expressed nvcomp compressed UCX buffers to be compressed after partitioning (on shuffle write) and after the shuffle read during our coalesce step, nothing else in the stack actually knows how to use this. In the spill framework, we make sure we are compatible with these buffers, but we don't take actions if the buffers are compressed. Here's the `TableMeta` for reference:

```
enum CodecType : byte {
  /// data simply copied, codec is only for testing
  COPY = -1,

  /// no compression codec was used on the data
  UNCOMPRESSED = 0,

  /// data compressed with the nvcomp LZ4 codec
  NVCOMP_LZ4 = 1,
}

/// Descriptor for a compressed buffer
table CodecBufferDescriptor {
  /// the compression codec used
  codec: CodecType;

  /// byte offset from the start of the enclosing compressed buffer
  /// where the compressed data begins
  compressed_offset: long;

  /// size of the compressed data in bytes
  compressed_size: long;

  /// byte offset from the start of the enclosing uncompressed buffer
  /// where the uncompressed data should be written
  uncompressed_offset: long;

  /// size of the uncompressed data in bytes
  uncompressed_size: long;
}

table BufferMeta {
  /// ID of this buffer
  id: int;

  /// size of the buffer data in bytes
  size: long;

  /// size of the uncompressed buffer data
  uncompressed_size: long;

  /// array of codec buffer descriptors if the data is compressed
  codec_buffer_descrs: [CodecBufferDescriptor];
}
```

Reply:
FYI. I filed an [issue](https://github.com/NVIDIA/spark-rapids/issues/9496) to follow up.